steel/docs/steel-history.md

```diff
@@ -0,0 +1,94 @@
+    .provider(provider.clone())
+    .block_number_or_tag(args.execution_block);
+
+#[cfg(any(feature = "beacon", feature = "history"))]
+let builder = builder.beacon_api(args.beacon_api_url);
+#[cfg(feature = "history")]
+let builder = builder.commitment_block(args.commitment_block);
+
+let mut env = builder.build().await?;
+
+// Preflight the call at the execution block.
+let mut contract = Contract::preflight(args.token_contract, &mut env);
+```
+
+The execution block is the block from which the view call state is retrieved. The commitment block is the block used for the Steel commitment. The commitment block has to fall within the 24 hour time window necessary for Steel commitment validation on-chain, but *crucially* the execution block can go further back on the scale of days, weeks or even months. 
+
+## How does Steel history work?
+
+The execution and commitment blocks are fundamentally related; the execution block should always be an ancestor of the commitment block. Therefore, it is possible to prove that the commitment block block root is canonical with the execution block block root by validating each beacon block root in between the two blocks.
+
+Steel handles this verification automatically when the history feature is enabled. Steel history works backwards from the commitment block to the execution block with consecutive calls to the beacon roots contract; validating a beacon root is a single call for every 24 hours of history and this step takes approximately 1M cycles within the Steel guest. 
+
+Ultimately, Steel will check the integrity of the view call data in the execution block by:
+
+1. proving that the execution block is a canonical ancestor of the commitment block
+2. carrying out the standard Steel commitment validation on-chain proving the integrity of the block root for the commitment block
+
+## How far can you go back?
+
+There is a hard limit on how far back in time you can place the execution block: the entire validation procedure depends on [EIP-4788] which was introduced with the [Cancun upgrade] on March 13 2024; this is the furthest that Steel history can go back.
+
+## How much does Steel history cost? 
+
+For the host, the developer needs to specify valid RPC URLs for both an archive execution node and a beacon node. You can see an example in the [publisher.rs] CLI args for the [erc20-counter] example.
+
+The greatest API cost will likely be from the beacon API endpoint. For each block between the commitment block and the execution block, Steel will query the full beacon block for verification. Please bear in mind that the wider the gap between the execution block and the commitment block, the larger the load on the beacon endpoint will be. 
+
+In terms of compute, the number of cycles for the Steel guest will also increase linearly with the number of blocks between the commitment block and the execution block. For every extra 24 hours of history, this is around 1 million cycles in the Steel guest.
```

Comment:
The cycle cost was before accelerators, when the keccak accelerator has landed it should significantly less expensive.

Reply:
thanks, added a note on that!